### PR TITLE
Rework audit_deps to use full advisory IDs (Uplift to 1.33.x)

### PR DESCRIPTION
**Pull request by diracdeltas**
Fix https://github.com/brave/brave-browser/issues/19592
Uplift to 1.33.x

### CODE REVIEW

**Review by wknapik — APPROVED** [review]

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.33.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
